File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int ans = -1;
        String res = "";
        int n = num.length();
        int i = 0;

        while (i < n) {
            if (1 <= i && i <= n - 2 && num.charAt(i - 1) == num.charAt(i) && num.charAt(i) == num.charAt(i + 1)) {
                int currentNum = Integer.parseInt(num.substring(i - 1, i + 2));
                if (ans < currentNum) {
                    ans = currentNum;
                    res = num.substring(i - 1, i + 2);
                }
            }
            i++;
        }

        return res;
    }
}

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        
        ans  = -1
        res = ""
        n = len(num)
        i=0
        while(i<n):

            if 1<=i<=n-2 and num[i-1] == num[i] and num[i] == num[i+1]:

                 if ans<int(num[i-1:i+2]) :
                    ans = int(num[i-1:i+2])
                    res = num[i-1:i+2]
         
            i+=1
            
        
        return res
    

            
        